kodi_addon_checker/check_addon_branches.py

```diff
@@ -0,0 +1,129 @@
+"""
+    Copyright (C) 2017-2018 Team Kodi
+    This file is part of Kodi - kodi.tv
+
+    SPDX-License-Identifier: GPL-3.0-only
+    See LICENSES/README.md for more information.
+"""
+
+import logging
+import os
+import xml.etree.ElementTree as ET
+
+from .check_dependencies import VERSION_ATTRB
+from .record import INFORMATION, PROBLEM, Record
+from .report import Report
+from .versions import AddonVersion, KodiVersion
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+def check_for_existing_addon(report: Report, addon_path: str, all_repo_addons: dict, pr: bool,
+                             kodi_version: KodiVersion):
+    """Check if addon submitted already exists or not
+        :addon_path: path of the addon
+        :all_repo_addons: dictionary return by all_repo_addon() function
+    """
+
+    addon_xml = os.path.join(addon_path, "addon.xml")
+    addon_name, addon_version = _get_addon_name(addon_xml)
+    addon_details = {'name': addon_name, 'version': addon_version}
+
+    is_new_addon = True
+    for branch, repo in sorted(all_repo_addons.items(), reverse=True):
+
+        # Addon submission must be higher than the versions already available in lower
+        # branches to allow users to receive the update (especially if run with --pr)
+        if KodiVersion(branch) <= kodi_version and addon_name in repo:
+            is_new_addon = False
+            _check_version_higher(report, addon_details, branch, repo.find(addon_name).version, pr)
+
+        # Addon submission must be lower than the versions already available in upper repo branches
+        # if that branch corresponds to a breaking change (e.g. version of addon in matrix version >
+        # version of addon in gotham) since xbmc.python is not backwards compatible.
+        elif KodiVersion(branch) > kodi_version and addon_name in repo and \
+            not _is_pythonabi_compatible(kodi_version, branch):
+            is_new_addon = False
+            _check_version_lower(report, addon_details, branch, repo.find(addon_name).version, pr)
+
+    if is_new_addon:
```

Comment:
I think you might now end up here for some cases that are not new addons?

Reply:
I was about to remove the `is_new_addon = False` from the above condition. The simple fact the addon is in a lower repository makes it not a new addon. So I think this is fine as is.